compiler/rustc_hir_analysis/src/coherence/inherent_impls_overlap.rs

```diff
@@ -65,8 +65,7 @@ impl<'tcx> InherentOverlapChecker<'tcx> {
     fn compare_hygienically(&self, item1: ty::AssocItem, item2: ty::AssocItem) -> bool {
         // Symbols and namespace match, compare hygienically.
         item1.namespace() == item2.namespace()
-            && item1.ident(self.tcx).normalize_to_macros_2_0()
-                == item2.ident(self.tcx).normalize_to_macros_2_0()
+            && ModernIdent::new(item1.ident(self.tcx)) == ModernIdent::new(item2.ident(self.tcx))
```

Comment:
I don't think replacing the `normalize_to_macros_2_0` calls in general is an improvement.
The new ident type makes sense only if it's stored in something or passed to something having an invariant.